lib/rex/socket/ssl_tcp.rb

```diff
@@ -123,10 +123,14 @@ def initsock_with_ssl_version(params, version)
     # Tie the context to a socket
     self.sslsock = OpenSSL::SSL::SSLSocket.new(self, self.sslctx)
 
-    # If peerhost looks like a hostname, set the undocumented 'hostname'
+    # If peerhostname is set, or if hostname looks like a hostname, set the undocumented 'hostname'
```

Comment:
```suggestion
    # If peerhostname is set, or if peerhost looks like a hostname, set the undocumented 'hostname'
```

Happy to ninja commit that doc string change.